File: arm.64/langs/fsharp/src/find.fs
namespace Pat

open System
open System.IO
open System.Linq

module Finder =
    let find (targetDir: string option) : string list =
        let dir =
            match targetDir with
            | Some d when not (String.IsNullOrEmpty d) -> d
            | _ ->
                let env = Environment.GetEnvironmentVariable "DIST_DIR"
                if not (String.IsNullOrEmpty env) then env
                else
                    let candidates = [
                        Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../../dist"))
                        Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../dist"))
                        Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(), "dist"))
                        "dist"
                    ]
                    candidates |> List.tryFind Directory.Exists |> Option.defaultValue "dist"

        if not (Directory.Exists dir) then []
        else
            Directory.GetFiles dir
            |> Array.filter (fun f -> Path.GetFileName f <> ".DS_Store")
            |> Array.toList

File: arm.64/langs/csharp/src/find.cs
namespace Pat;

public static class Finder
{
    public static List<string> Find(string? targetDir = null)
    {
        string? dir = targetDir;
        if (string.IsNullOrEmpty(dir))
        {
            dir = Environment.GetEnvironmentVariable("DIST_DIR");
            if (string.IsNullOrEmpty(dir))
            {
                string[] candidates = {
                    Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../../dist")),
                    Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "../../dist")),
                    Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(), "dist")),
                    "dist"
                };
                dir = candidates.FirstOrDefault(Directory.Exists) ?? "dist";
            }
        }

        if (!Directory.Exists(dir)) return new List<string>();

        return Directory.GetFiles(dir)
            .Where(f => Path.GetFileName(f) != ".DS_Store")
            .ToList();
    }
}
